src/computability/turing_machine.lean

```diff
@@ -654,10 +654,21 @@ theorem reaches₀.tail' {σ} {f : σ → option σ} {a b c : σ}
   (h : reaches₀ f a b) (h₂ : c ∈ f b) : reaches₁ f a c :=
 h _ (trans_gen.single h₂)
 
+theorem eval_match_lemma {σ} {a : σ} {a' b} :
+  sum.inr b ∈ roption.some (eval._match_1 a a') ↔ a' = some b :=
```

Comment:
I really don't like lemmas that mention `_match_*` by name.  How about removing the match in `eval`?
```lean
def eval {σ} (f : σ → option σ) : σ → roption σ :=
pfun.fix (λ s, roption.some $ (f s).elim (sum.inl s) sum.inr)
```

Reply:
I think I don't actually use this lemma anyway (it dates from a previous version of the proof), so I've removed it and used `option.elim` instead of a match.